OutaTime/OutaTime/ViewControllers/TimeCircuitsViewController.swift

```diff
@@ -0,0 +1,96 @@
+    
+    var dateFormatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MMM dd yyyy"
+        return formatter
+    }
+    
+    var speed: Int = 0
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        destinationTimeLabel.text = "--- -- ----"
+        presentTimeLabel.text = dateFormatter.string(from: Date())
+        lastTimeDepartedLabel.text = "--- -- ----"
+        speedLabel.text = "\(speed) MPH"
+        travelBackButton.setTitleColor(UIColor.darkGray, for: .normal)
+    }
+    
+    private var timer: Timer? = nil
+    
+    func startTimer() {
+        travelBackButton.setTitleColor(UIColor.darkGray, for: .normal)
+        travelBackButton.isEnabled = false
+        timer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true, block: updateSpeed(timer:))
+    }
+        
+    func resetTimer() {
+        timer?.invalidate()
+        timer = nil
+    }
+        
+    private func updateSpeed(timer: Timer) {
+        if speed < 88 {
+            speed += 1
+            speedLabel.text = "\(speed) MPH"
+        } else {
+            resetTimer()
+            travelBackButton.isEnabled = true
+            lastTimeDepartedLabel.text = presentTimeLabel.text
+            presentTimeLabel.text = destinationTimeLabel.text
+            destinationTimeLabel.text = "--- -- ----"
+            speed = 0
+            speedLabel.text = "\(speed) MPH"
+            let alert = UIAlertController(title: "Time Travel Successful", message: "Your new date is \(presentTimeLabel.text ?? "unknown")", preferredStyle: .alert)
+            let okButton = UIAlertAction(title: "OK", style: .cancel, handler: nil)
+            alert.addAction(okButton)
+            present(alert, animated: true, completion: nil)
+        }
+    }
+    
+    @IBAction func travelBackButton(_ sender: Any) {
+        if destinationTimeLabel.text != "--- -- ----" {
+            startTimer()
+        } else {
+            let alert = UIAlertController(title: "No Destination Time", message: "Please set a destination time.", preferredStyle: .alert)
+            let okButton = UIAlertAction(title: "OK", style: .cancel, handler: nil)
+            alert.addAction(okButton)
+            present(alert, animated: true, completion: nil)
+        }
+    }
+    
+    // MARK: - Navigation
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "modalDestinationDatePickerSegue" {
+            if let datePickerVC = segue.destination as? DatePickerViewController {
+                datePickerVC.delegate = self
+            }
+        }
+    }
+
+}
+
+extension TimeCircuitsViewController: DatePickerDelegate {
+    func destinationDateWasChosen(_ date: Date) {
+        destinationTimeLabel.text = dateFormatter.string(from: date)
+        travelBackButton.setTitleColor(UIColor(red:83/255, green:27/255, blue:147/255, alpha:1.0), for: .normal)
```

Comment:
Nice use of a custom color :) If a color is going to be used throughout an app, you can put it in a constants file. So you could have a file Constants.swift, and then in there have all your constants such as fonts, colors, etc... One great way to organize them and keep them from clogging up the global namespace is to put constants in a case-less enum. That would look like this:

``` swift
enum Color {
    static let myColor = UIColor(red:83/255, green:27/255, blue:147/255, alpha:1.0)
}
```

You would then access that color anywhere in your app by typing Color.myColor

Another popular approach is to use an extension on UIColor:

``` swift
extension UIColor {
    static let myColor = UIColor(red:83/255, green:27/255, blue:147/255, alpha:1.0)
}
```

This has the advantage of allowing you to simply type a dot and then the name of your color when you need to use it, eg:

``` swift
travelBackButton.setTitleColor(.myColor, for: .normal)
```

Sorry for going down a tangent, as in this case it's fine to declare the color in line as you have done, just letting you know of options for future apps that are larger.